File: BOJ/Python/그리디 알고리즘/1138.py
N = int(input())
arr = list(map(int, input().split()))
res = [0 for x in range(N)]

for i in range(1, N+1):
    temp = arr[i-1]
    count = 0

    for j in range(N):
        if count == temp and res[j] == 0:
            res[j] = i
            break
        if res[j] == 0:
            count += 1

for i in res:
    print(i, end=' ')

File: BOJ/Kotlin/그리디 알고리즘/1138.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val N = readLine().toInt()
    val arr = readLine().split(' ').map { it.toInt() }
    val res = arrayOfNulls<Int>(N)

    for (i in 1..N){
        val temp = arr[i-1]
        var count = 0

        for (j in 0 until N){
            if (count == temp && res[j] == null){
                res[j] = i
                break
            }
            if (res[j] == null){
                count ++
            }
        }
    }

    res.forEach { print("$it ") }
}